assets/src/apps/authoring/components/RightMenu/RightMenu.tsx

```diff
@@ -302,7 +304,28 @@ const RightMenu: React.FC<any> = () => {
     },
     [currentActivity, currentPartInstance, currentPartSelection],
   );
-
+  const handleEditComponentJson = (newJson: any, isCancelled = false) => {
+    setDisplayEditor(false);
+    if(isCancelled)
+      return;
+    const cloneActivity = clone(currentActivity);
+    const ogPart = cloneActivity.content?.partsLayout.find((part: any) => part.id === currentPartSelection);
+    if (!ogPart) {
+      console.warn(
+        'couldnt find part in current activity, most like lives on a layer; you need to update they layer copy directly',
+      );
+      return;
+    }
+    if (newJson.id !== ogPart.id) {
+      ogPart.id = newJson.id;
+      // in case the id changes, update the selection
+      dispatch(setCurrentSelection({ selection: newJson.id }));
+    }
+    ogPart.custom = newJson.custom;
```

Comment:
we need to do some kind of validation here, since we don't have the form fields from the property editor restricting what they can input. perhaps like above we should also pass into the component the schema data (or just use it in the component) and when it does the parse test on update make sure that they don't delete any required fields. also need to make sure that the same thing with the palette happens, so need to make sure that `useHTMLProps` gets set to true, and it has the `backgroundColor` and friends properties.